File: 2583-kth-largest-sum-in-a-binary-tree/2583-kth-largest-sum-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void bfs(TreeNode *root,vector<long long> &v){
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int sz = q.size();
            long long sum = 0;
            for(int i=0;i<sz;i++){
                TreeNode *node = q.front();
                q.pop();
                sum += node->val;
                if(node->left)
                    q.push(node->left);
                if(node->right)
                    q.push(node->right);
            }
            v.push_back(sum);
        }
    }
    
    long long kthLargestLevelSum(TreeNode* root, int k) {
        vector<long long> v;
        bfs(root,v);
        sort(v.begin(),v.end());
        if(k > v.size())
            return -1;
        return v[v.size()-k];
    }
};

File: 2583-kth-largest-sum-in-a-binary-tree/2583-kth-largest-sum-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void bfs(TreeNode root,ArrayList<Long> v){
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        
        while(!q.isEmpty()){
            int sz = q.size();
            long sum = 0;
            for(int i=0;i<sz;i++){
                TreeNode node = q.peek();
                q.poll();
                sum += node.val;
                if(node.left != null)
                    q.offer(node.left);
                if(node.right != null)
                    q.offer(node.right);
            }
            v.add(sum);
        }
    }
    
    public long kthLargestLevelSum(TreeNode root, int k) {
        ArrayList<Long> v = new ArrayList<>();
        bfs(root,v);
        if(k > v.size())
            return -1;
        
        Collections.sort(v,(a,b)->b.compareTo(a)); // descending order sort
        return v.get(k-1);
    }
}